lib/api/apiUtils/bucket/bucketWebsite.js

```diff
@@ -0,0 +1,275 @@
+        if (_xmlContainsElem(parent, 'Protocol')) {
+            if (parent[0].Protocol[0] !== 'http' &&
+            parent[0].Protocol[0] !== 'https') {
+                errMsg = 'Invalid protocol, protocol can be http or https. ' +
+                'If not defined, the protocol will be selected automatically.';
+                return errors.InvalidRequest.customizeDescription(errMsg);
+            }
+            redirectAllObj.protocol = parent[0].Protocol[0];
+        }
+        websiteConfig.setRedirectAllRequestsTo(redirectAllObj);
+    }
+
+    if (resultConfig.IndexDocument) {
+        const parent = resultConfig.IndexDocument;
+        if (!_xmlContainsElem(parent, 'Suffix')) {
+            errMsg = 'IndexDocument is not well-formed';
+            return errors.MalformedXML.customizeDescription(errMsg);
+        } else if (typeof parent[0].Suffix[0] !== 'string'
+        || parent[0].Suffix[0] === ''
+        || parent[0].Suffix[0].indexOf('/') !== -1) {
+            errMsg = 'IndexDocument Suffix is not well-formed';
+            return errors.InvalidArgument.customizeDescription(errMsg);
+        }
+        websiteConfig.setIndexDocument(parent[0].Suffix[0]);
+    }
+
+    if (resultConfig.ErrorDocument) {
+        const parent = resultConfig.ErrorDocument;
+        if (!_xmlContainsElem(parent, 'Key')) {
+            errMsg = 'ErrorDocument is not well-formed';
+            return errors.MalformedXML.customizeDescription(errMsg);
+        }
+        if (typeof parent[0].Key[0] !== 'string'
+        || parent[0].Key[0] === '') {
+            errMsg = 'ErrorDocument Key is not well-formed';
+            return errors.InvalidArgument.customizeDescription(errMsg);
+        }
+        websiteConfig.setErrorDocument(parent[0].Key[0]);
+    }
+
+    if (resultConfig.RoutingRules) {
+        const routingRule = new RoutingRule();
+        const parent = resultConfig.RoutingRules;
+        if (!_xmlContainsElem(parent, 'RoutingRule', true)) {
+            errMsg = 'RoutingRules is not well-formed';
+            return errors.MalformedXML.customizeDescription(errMsg);
+        }
+        for (let i = 0; i < parent[0].RoutingRule.length; i++) {
+            const rule = parent[0].RoutingRule[i];
+            const redirectObj = {};
+            if (!(Array.isArray(rule.Redirect) && rule.Redirect.length === 1)) {
+                errMsg = 'RoutingRule requires Redirect, which is ' +
+                'missing or not well-formed';
+                return errors.MalformedXML.customizeDescription(errMsg);
+            }
+            // looks like AWS doesn't actually make this check:
+            if (!(_xmlContainsElem(rule.Redirect, 'Protocol')
+            || _xmlContainsElem(rule.Redirect, 'HostName')
+            || _xmlContainsElem(rule.Redirect, 'ReplaceKeyPrefixWith')
+            || _xmlContainsElem(rule.Redirect, 'ReplaceKeyWith')
+            || _xmlContainsElem(rule.Redirect, 'HttpRedirectCode'))) {
```

Comment:
I extended xmlContainsElem to handle multiple elements, but the logic of the function unfortunately got a bit more complicated. :sweat_smile: